ItemLevelDownloader/Engine/artifactEngine.ts

```diff
@@ -0,0 +1,104 @@
+﻿import { setTimeout } from 'timers';
+import * as path from 'path';
+import * as fs from 'fs';
+
+var minimatch = require('minimatch');
+
+import * as models from '../Models';
+import { ArtifactItemStore } from '../Store/artifactItemStore';
+import { ArtifactEngineOptions } from "./artifactEngineOptions"
+
+export class ArtifactEngine {
+    async processItems(sourceProvider: models.IArtifactProvider, destProvider: models.IArtifactProvider, artifactEngineOptions?: ArtifactEngineOptions): Promise<void> {
+        const processors: Promise<{}>[] = [];
+        artifactEngineOptions = artifactEngineOptions || new ArtifactEngineOptions();
+        this.artifactItemStore.flush();
```

Comment:
this causes problem when i sequence or chain processItems(). e.g. I can call `engine.processItems(<jenkins-provider>, <local-file-provider>).processItems(<local-file-provider>, <azure-blobs-provider>)`